### DISCUSSION TITLE
Local dependencies in [[metadata.dependencies]]

### DISCUSSION THREAD

**Discussion by yeachan153** [paketo-buildpacks/feedback#193] [Q&A] [answered]
Hi all,

Does anyone know if it's possible to provide a local file path to the URI in the `[[metadata.dependencies]]` of `buildpack.toml`? I've placed the file there in my local file system and entered the path of the local tar there: `/tmp/bla.tar.gz`, but unfortunately I get an error during packing:

```bash
Error: failed to cache dependencies: failed to download dependency: failed to make request: Get "/tmp/bla.tar.gz": unsupported protocol scheme ""
```

Thank you

**Comment by sophiewigmore** [comment, marked as answer]
Yes you can, you need to prefix the filepath with `file://` so it looks like: `uri: "file://<path to file>`. Then you can copy in the file to the top-level of the buildpack dir, added the file name to `include-files` array at the top of the `buildpack.toml`.

**Reply by yeachan153** [reply]
awesome, thanks

**Comment by dmikusa** [comment]
Buildpacks written with packit and libpak also have the capability to reference remote URLs but bundle the dependencies with the buildpack at package time (often called bundling in offline mode). This tends to be more flexible, cause your buildpack can be published in both ways since it's a package time decision whether you want dependencies include or not.

With offline mode, the net result is basically the same as what @sophiewigmore mentioned.
